planetary-social/nos
# Use NIP-92 media metadata to display media in the proper orientation

## Issues covered
#1172 

## Description
Gets the media metadata from the `imeta` tag, if it exists, and uses the dimensions to determine whether the gallery should be displayed in landscape or portrait.

- [x] Fix the bad code
- [x] Deduplicate code in GalleryView (inside `loadingView(_:)` where we check the metadata for orientation twice)

## How to test
There's no visible difference here. The big change is that we're using `imeta` tags from the event to determine orientation of the images/video before loading the images/video. So maybe you'll want to use [everyone's favorite proxy tool](https://proxyman.io) and the debugger to verify.

1. Navigate to Settings
2. Enable new media display
3. Navigate to Feed or Profile
4. Find some media with an `imeta` tag (`note1cj6xmz48ausu04mckl4ral8qec4j4sz7akr4uarewwchsydywraq8jwkt8`)
5. See if it's displayed correctly based on the tag

I posted some images from Damus since it includes `imeta` tags, then stepped through the debugger to be sure we're using the `imeta` tags to determine orientation instead of loading the image and checking its size to determine orientation.

## Screenshots/Video
<img width="201" alt="Screenshot 2024-09-04 at 4 56 02 PM" src="https://github.com/user-attachments/assets/6958d4a1-55a1-4616-bffc-be7bd412b94a">
